python/CMakeLists.txt

```diff
@@ -41,38 +41,41 @@ if (PYTHON_EXECUTABLE)
     set(__configs ${CMAKE_CONFIGURATION_TYPES})
   endif()
   foreach (__config ${__configs})
-    set(CONFIG ${__configs})
-    if(${__configs} STREQUAL "Debug")
+    set(CONFIG ${__config})
+    if(${__config} STREQUAL "Debug")
       set(PYTHON_PACKAGE_NAME "_phylanxd")
     else()
       set(PYTHON_PACKAGE_NAME "_phylanx")
     endif()
-    configure_file(${SETUP_PY_IN} "${SETUP_PY}_${__configs}.py" @ONLY)
+    configure_file(${SETUP_PY_IN} "${SETUP_PY}_${__config}.py" @ONLY)
   endforeach()
 
   configure_file(${INIT_PY_IN} "${INIT_PY}" @ONLY)
 
-  set(__setup_script "${SETUP_PY}_${__configs}.py")
-  set(setup_files ${SETUP_PY_IN} ${INIT_PY_IN})
 
-  add_custom_command(OUTPUT ${OUTPUT}
-    COMMAND ${PYTHON_EXECUTABLE} ${__setup_script} build
-    COMMAND ${CMAKE_COMMAND} -E touch ${OUTPUT}
-    DEPENDS ${DEPS} phylanx_py)
+  foreach (__config ${__configs})
+    set(__setup_script "${SETUP_PY}_${__config}.py")
+    set(setup_files ${SETUP_PY_IN} ${INIT_PY_IN})
+
+    add_custom_command(OUTPUT ${OUTPUT}
+      COMMAND ${PYTHON_EXECUTABLE} ${__setup_script} build 
+      COMMAND ${CMAKE_COMMAND} -E touch ${OUTPUT}
+      DEPENDS ${DEPS} phylanx_py)
 
-  add_custom_target(python_setup ALL
-    DEPENDS ${OUTPUT} ${__setup_script} ${DEPS}
-    SOURCES ${setup_files})
+    add_custom_target(python_setup ALL
+      DEPENDS ${OUTPUT} ${__setup_script} ${DEPS}
+      SOURCES ${setup_files})
 
-  set_target_properties(python_setup
-    PROPERTIES FOLDER "Core")
+    set_target_properties(python_setup
+      PROPERTIES FOLDER "Core")
 
-  source_group("Setup" FILES ${setup_files})
-  set_source_files_properties(${setup_files} PROPERTIES HEADER_FILE_ONLY TRUE)
+    source_group("Setup" FILES ${setup_files})
+    set_source_files_properties(${setup_files} PROPERTIES HEADER_FILE_ONLY TRUE)
 
-  # for the install step it does not matter which setup.py we use
-  install(CODE
-    "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY}_${__configs}.py install --user)")
+    # for the install step it does not matter which setup.py we use
+    install(CODE
+      "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY}_${__config}.py install --user)")
+  endforeach()
```

Comment:
I think the installation does not need to be performed more than once. The initial idea was to execute only the `setup_${__config}.py` script for one of the available configuration targets as the generated scripts all do the same thing for the installation (they differ for the build step only) - hence the comment...

Reply:
I agree, but I figured there was no harm. In any event, I've pushed a change that only runs it once. I think that my changes have nothing to do with the failing tests, however.

Reply:
The failing C++ tests are a (known) HPX problem, please disregard those. The Python tests however seem to run fine.